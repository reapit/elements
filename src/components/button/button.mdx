import { Meta, Story, Canvas, Controls, Description } from '@storybook/blocks'
import { Button } from './index'
import { RenderHtmlMarkup } from '../../storybook/render-html-markup'
import * as ButtonStories from './button.stories'

<Meta of={ButtonStories} />

# Button

<Controls />

Styled button component with multiple flavours depending on use case.

Props available for the React version of the component;

## Default Usage

A button with no addtional props / classes applied.

<Canvas of={ButtonStories.DefaultUsage} />

<RenderHtmlMarkup of={ButtonStories.DefaultUsage} />

## With Intent

Different intents are supported, to enable buttons to be used in different contexts. Typcially you will always choose an intent for your button. The intent prop maps directly to the intent helper classes for non React users.

Whereas most components support the full range of intents, `Buttons` only support `primary`, `danger` and `default`, the latter being the default.

<Canvas of={ButtonStories.Intent} />

<RenderHtmlMarkup of={ButtonStories.Intent} />

## Is Loading

Add a loading spinner to indicate progress of a background activity after the button was clicked.

<Canvas of={ButtonStories.Loading} />

<RenderHtmlMarkup of={ButtonStories.Loading} />

## Is Disabled

Indicate that a button is disabled for whatever reason.

<Canvas of={ButtonStories.Disabled} />

<RenderHtmlMarkup of={ButtonStories.Disabled}/>

## Other attributes

All other standard HTML attributes for `<button>` are supported and are passed through to the React component, e.g. type and onClick

<Canvas of={ButtonStories.StandardAttributes} />

<RenderHtmlMarkup of={ButtonStories.StandardAttributes} />

## Button Group

An element to collectively group several buttons

<Description of={ButtonStories.DeprecatedGroup} />

<Canvas of={ButtonStories.DeprecatedGroup} />

<RenderHtmlMarkup of={ButtonStories.DeprecatedGroup} />

## Button Size

If you wish to size up or down a button, you can use the `buttonSize` prop.

By default all of our buttons are sized to `small` in mobile and `medium` in desktop so this option is solely if you wish to force the size.

<Canvas of={ButtonStories.ButtonSize}/>

<RenderHtmlMarkup of={ButtonStories.ButtonSize}/>

## Button Icons

If you wish to add a system icon from the `IconList`, you can do this by passing the `buttonIcon` prop.

You have the option to specify a position of `left`, `right` or `only` as a postion for your icon.

<Canvas of={ButtonStories.ButtonIcons}/>

<RenderHtmlMarkup of={ButtonStories.ButtonIcons} />

## Floating Button

A circular floating icon button - supports any icon from the icon list.

Props available for the React version of the component are the same as the main button.

<Canvas of={ButtonStories.Floating}/>

<RenderHtmlMarkup of={ButtonStories.Floating} />
