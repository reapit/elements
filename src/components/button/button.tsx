import {
  HTMLAttributes,
  AnchorHTMLAttributes,
  ButtonHTMLAttributes,
  FC,
  MouseEvent,
  MouseEventHandler,
  ReactNode,
} from 'react'
import {
  ElButton,
  elButtonSizeLarge,
  elButtonSizeMedium,
  elButtonSizeSmall,
  elButtonIconOnly,
  ElButtonSpinner,
  ElAnchorButton,
  elFloatingButton,
  DeprecatedElButtonGroup,
  ElButtonGroupInner,
  elButtonGroupAlignLeft,
  elButtonGroupAlignRight,
  elButtonGroupAlignCenter,
  ElButtonLabel,
} from './styles'
import { Icon, IconNames } from '../icon'
import { cx } from '@linaria/core'

type ButtonSize = 'small' | 'medium' | 'large'

type ButtonAsButtonVariant = 'primary' | 'secondary' | 'tertiary' | 'destructive' | 'busy'
type ButtonAsAnchorVariant = Exclude<ButtonAsButtonVariant, 'busy'>

interface CommonButtonProps {
  children?: ReactNode
  variant?: ButtonAsButtonVariant
  size?: ButtonSize
  iconLeft?: ReactNode
  iconRight?: ReactNode
  /** The label for button. It must be supplied for buttons with no `children` */
  'aria-label'?: string
  className?: string
}

interface ButtonAsButtonElementProps extends CommonButtonProps, ButtonHTMLAttributes<HTMLButtonElement> {
  href?: never
  target?: never
  disabled?: boolean
  onClick?: MouseEventHandler<HTMLButtonElement>
}

interface ButtonAsAnchorElementProps extends CommonButtonProps, AnchorHTMLAttributes<HTMLAnchorElement> {
  variant?: ButtonAsAnchorVariant
  /** Button styled <a> element should always have href */
  href: string
  target?: string
  rel?: string
  /** Anchor elements cannot be disabled. Use a button element if the component needs to be in a disabled state */
  disabled?: never
  onClick?: MouseEventHandler<HTMLAnchorElement>
}

export type ButtonProps = ButtonAsButtonElementProps | ButtonAsAnchorElementProps

function isButtonAsButtonElement(props: ButtonProps): props is ButtonAsButtonElementProps {
  return !props.href
}

export const handleButtonClick =
  (onClick?: MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>) =>
  (e: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => {
    if (onClick) onClick(e)
  }

/** @deprecated */
export interface FloatingButtonProps extends ButtonAsButtonElementProps {
  icon: IconNames
}

export type ButtonGroupAlignment = 'left' | 'right' | 'center'

/**
 * A subset of button, button group should not be used.
 * @deprecated Will be removed in future major version. Use `button-group` as parent component.
 */
export interface ButtonGroupProps extends HTMLAttributes<HTMLDivElement> {
  alignment?: ButtonGroupAlignment
}

export const Button: FC<ButtonProps> = ({
  children,
  variant,
  size = 'medium',
  iconLeft,
  iconRight,
  onClick,
  'aria-label': ariaLabel,
  disabled = false,
  href,
  target,
  rel,
  className,
  ...rest
}) => {
  const sizeClass = cx(
    size === 'small' && elButtonSizeSmall,
    size === 'medium' && elButtonSizeMedium,
    size === 'large' && elButtonSizeLarge,
  )

  const miscellaneousClass = cx(
    !children && elButtonIconOnly, // if no children(label) then add el-button-icon-only class
  )

  const combinedClassName = cx(className, sizeClass, miscellaneousClass)

  if (!isButtonAsButtonElement({ href })) {
    return (
      <ElAnchorButton
        href={href}
        data-variant={variant}
        className={combinedClassName}
        onClick={(e) => {
          e.preventDefault()
          handleButtonClick(onClick)
        }}
        aria-label={ariaLabel}
        role="button"
        target={target}
        rel={rel}
        {...(rest as AnchorHTMLAttributes<HTMLAnchorElement>)}
      >
        <ElButtonSpinner />
        {variant !== 'busy' && iconLeft}
        {children && <ElButtonLabel>{children}</ElButtonLabel>}
        {variant !== 'busy' && iconRight}
      </ElAnchorButton>
    )
  } else {
    return (
      <ElButton
        data-variant={variant}
        className={combinedClassName}
        onClick={handleButtonClick(onClick)}
        aria-label={ariaLabel}
        aria-disabled={disabled}
        role="button"
        {...(rest as ButtonHTMLAttributes<HTMLButtonElement>)}
      >
        <ElButtonSpinner />
        {variant !== 'busy' && iconLeft}
        {children && <ElButtonLabel>{children}</ElButtonLabel>}
        {variant !== 'busy' && iconRight}
      </ElButton>
    )
  }
}

/**
 * Moved to its own component `ButtonGroup`.
 * @deprecated `alignment` prop will not be supported in new ButtonGroup component.
 */
export const DeprecatedButtonGroup: FC<ButtonGroupProps> = ({ children, alignment, ...rest }) => {
  const alignmentClass = cx(
    alignment === 'left' && elButtonGroupAlignLeft,
    alignment === 'right' && elButtonGroupAlignRight,
    alignment === 'center' && elButtonGroupAlignCenter,
  )
  return (
    <DeprecatedElButtonGroup {...rest}>
      <ElButtonGroupInner className={alignmentClass}>{children}</ElButtonGroupInner>
    </DeprecatedElButtonGroup>
  )
}

/** @deprecated */
// Removing ClassName fixes the issue with different UI, lets look into this
export const FloatingButton: FC<FloatingButtonProps> = ({ className, icon, ...rest }) => {
  return <Button className={cx(className, elFloatingButton)} iconLeft={<Icon icon={icon} />} {...rest} />
}
