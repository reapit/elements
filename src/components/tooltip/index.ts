export * from './styles'
export * from './tooltip'
