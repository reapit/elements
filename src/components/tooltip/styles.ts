import { styled } from '@linaria/react'

export const ElTooltip = styled.div`
  width: max-content;
  background: var(--fill-default-darkest, #222b33);
  color: var(--text-white);
  padding: var(--spacing-2) var(--spacing-3) var(--spacing-2) var(--spacing-3);
  border-radius: var(--corner-default);
  font-family: var(--font-family);
  font-size: var(--font-size-xs);
  font-weight: var(--font-weight-regular);
  line-height: var(--line-height-xs);
  letter-spacing: var(--letter-spacing-xs);
  text-align: left;
  position: absolute;
  transition: opacity 0.2s;
  pointer-events: none;
  white-space: normal;
  word-wrap: break-word;
  z-index: 99999; // Adding additional CSS to position tooltip on top of everything

  // To do (In Future): Tooltip Positioning to be replaced with css anchor positioning feature
  // Currently not supported by wider browser group
  // See: https://developer.mozilla.org/en-US/docs/Web/CSS/position-anchor#browser_compatibility
`

export const ElTooltipLabel = styled.span`
  font-family: var(--font-family);
  font-size: var(--font-size-xs);
  font-weight: var(--font-weight-semibold);
  line-height: var(--line-height-xs);
  letter-spacing: var(--letter-spacing-xs);
  text-align: left;
`
