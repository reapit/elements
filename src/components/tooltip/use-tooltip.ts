import { HTMLAttributes, useEffect, useState } from 'react'
import { useId } from '#src/storybook/random-id'

export const useTooltip = () => {
  const tooltipId = `tooltip-id-${useId()}`
  const [isVisible, setIsVisible] = useState(false)

  const show = () => setIsVisible(true)
  const hide = () => setIsVisible(false)

  const positionTooltip = () => {
    // To do (In Future): Tooltip/Trigger selector to use useRef once project upgraded to React 19
    // See: https://react.dev/reference/react/forwardRef
    // See: https://react.dev/blog/2024/12/05/react-19#ref-as-a-prop
    const tooltip = document.querySelector<HTMLDivElement>(`[id="${tooltipId}"]`)
    const trigger = document.querySelector<HTMLElement>(`[data-visible-id="${tooltipId}"]`)

    if (!tooltip || !trigger) return

    const triggerRect = trigger.getBoundingClientRect()
    const tooltipRect = tooltip.getBoundingClientRect()

    // Calculate scroll offsets
    const scrollTop = document.documentElement.scrollTop || document.body.scrollTop
    const scrollLeft = document.documentElement.scrollLeft || document.body.scrollLeft

    // const viewportWidth = window.innerWidth
    // const viewportHeight = window.innerHeight
    const position = tooltip.getAttribute('data-position')

    let top = 0
    let left = 0

    switch (position) {
      case 'top':
        top = triggerRect.top + scrollTop - tooltipRect.height - 4
        left = triggerRect.left + scrollLeft + (triggerRect.width - tooltipRect.width) / 2
        break
      case 'top-start':
        top = triggerRect.top + scrollTop - tooltipRect.height - 4
        left = triggerRect.left + scrollLeft
        break
      case 'top-end':
        top = triggerRect.top + scrollTop - tooltipRect.height - 4
        left = triggerRect.right + scrollLeft - tooltipRect.width
        break
      case 'bottom':
        top = triggerRect.bottom + scrollTop + 4
        left = triggerRect.left + scrollLeft + (triggerRect.width - tooltipRect.width) / 2
        break
      case 'bottom-start':
        top = triggerRect.bottom + scrollTop + 4
        left = triggerRect.left + scrollLeft
        break
      case 'bottom-end':
        top = triggerRect.bottom + scrollTop + 4
        left = triggerRect.right + scrollLeft - tooltipRect.width
        break
      case 'left':
        top = triggerRect.top + scrollTop + (triggerRect.height - tooltipRect.height) / 2
        left = triggerRect.left + scrollLeft - tooltipRect.width - 4
        break
      case 'left-start':
        top = triggerRect.top + scrollTop
        left = triggerRect.left + scrollLeft - tooltipRect.width - 4
        break
      case 'left-end':
        top = triggerRect.bottom + scrollTop - tooltipRect.height
        left = triggerRect.left + scrollLeft - tooltipRect.width - 4
        break
      case 'right':
        top = triggerRect.top + scrollTop + (triggerRect.height - tooltipRect.height) / 2
        left = triggerRect.right + scrollLeft + 4
        break
      case 'right-start':
        top = triggerRect.top + scrollTop
        left = triggerRect.right + scrollLeft + 4
        break
      case 'right-end':
        top = triggerRect.bottom + scrollTop - tooltipRect.height
        left = triggerRect.right + scrollLeft + 4
        break
      default:
        break
    }

    // Below are inline style applied to tooltip to position it relevant to respective trigger
    tooltip.style.position = 'absolute'
    tooltip.style.inset = '0px auto auto 0px'
    tooltip.style.transform = `translate(${left}px, ${top}px)`
    tooltip.style.margin = '0px'
  }

  // Update tooltip position on visibility change or window resize
  useEffect(() => {
    if (isVisible) {
      positionTooltip()
      window.addEventListener('resize', positionTooltip)
    }

    return () => {
      window.removeEventListener('resize', positionTooltip)
    }
  }, [isVisible])

  const getTriggerProps = (props?: HTMLAttributes<HTMLElement>) => ({
    ...props,
    'data-visible-id': tooltipId,
    'aria-describedby': tooltipId,
    onFocus: (e: React.FocusEvent<HTMLElement>) => {
      props?.onFocus?.(e)
      show()
    },
    onBlur: (e: React.FocusEvent<HTMLElement>) => {
      props?.onBlur?.(e)
      hide()
    },
    onMouseEnter: (e: React.MouseEvent<HTMLElement>) => {
      props?.onMouseEnter?.(e)
      show()
    },
    onMouseLeave: (e: React.MouseEvent<HTMLElement>) => {
      props?.onMouseLeave?.(e)
      hide()
    },
  })

  const getTooltipProps = (props?: HTMLAttributes<HTMLDivElement>) => ({
    ...props,
    id: tooltipId,
    role: 'tooltip',
    'aria-hidden': !isVisible,
    isVisible,
  })

  return { getTriggerProps, getTooltipProps }
}
