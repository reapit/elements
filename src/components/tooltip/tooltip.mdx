import { Meta, Canvas, Controls, Description } from '@storybook/blocks'
import { RenderHtmlMarkup } from '../../storybook/render-html-markup'
import * as TooltipStories from './tooltip.stories'

<Meta of={TooltipStories} />

# Tooltip

<Description of={TooltipStories.DisplayTooltipWithoutTrigger} />
<Canvas of={TooltipStories.DisplayTooltipWithoutTrigger} />
<RenderHtmlMarkup of={TooltipStories.DisplayTooltipWithoutTrigger} />

<Controls />

<Description of={TooltipStories.BasicUsage} />
<Canvas of={TooltipStories.BasicUsage} />
