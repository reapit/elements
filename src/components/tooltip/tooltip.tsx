import React, { FC, HTMLAttributes } from 'react'
import { ElTooltip, ElTooltipLabel } from './styles'
import { createPortal } from 'react-dom'

export interface TooltipProps extends HTMLAttributes<HTMLDivElement> {
  label?: string
  description: string
  isVisible?: boolean
  maxWidth?: string
  position?:
    | 'top'
    | 'bottom'
    | 'right'
    | 'left'
    | 'top-start'
    | 'top-end'
    | 'bottom-start'
    | 'bottom-end'
    | 'right-start'
    | 'right-end'
    | 'left-start'
    | 'left-end'
}

export const Tooltip: FC<TooltipProps> = ({
  isVisible,
  label,
  description,
  maxWidth = '400px',
  position = 'top',
  ...rest
}) => {
  const tooltip = (
    <ElTooltip role="tooltip" data-position={position} style={{ maxWidth: maxWidth }} {...rest}>
      {label && <ElTooltipLabel>{label}: </ElTooltipLabel>}
      {description}
    </ElTooltip>
  )
  if (!isVisible) {
    return null
  }
  return createPortal(tooltip, document.body)
}
