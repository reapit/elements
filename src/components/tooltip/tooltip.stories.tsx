import { Meta } from '@storybook/react'
import { Button } from '../button'
import { useTooltip } from './use-tooltip'
import { Tooltip, TooltipProps } from './tooltip'
import { ElTooltip, ElTooltipLabel } from './styles'

const meta: Meta<typeof Tooltip> = {
  title: 'Components/Tooltip',
  component: Tooltip,
  argTypes: {
    description: {
      control: 'text',
      description: 'Defines the description of the tooltip.',
    },
    label: {
      control: 'text',
      description: 'Defines the label of the tooltip',
    },
    position: {
      control: 'select',
      options: [
        'top',
        'bottom',
        'right',
        'left',
        'top-start',
        'top-end',
        'bottom-start',
        'bottom-end',
        'right-start',
        'right-end',
        'left-start',
        'left-end',
      ],
      description: 'Defines where to position of the tooltip relative to the trigger.',
      table: {
        type: { summary: 'enum' },
        defaultValue: { summary: 'top' },
      },
    },
    maxWidth: {
      control: 'text',
      description: 'Defines the max-width of the tooltip container.',
      table: {
        type: { summary: 'string', detail: 'CSS max-width value (e.g., 400px, 50%)' },
        defaultValue: { summary: '400px' },
      },
    },
  },
}

export default meta

/**
 * The Tooltip component is a lightweight UI element used to display contextual
 * information when a user hovers over or focuses on a target element.
 * In the default configuration, the tooltip provides a brief label and/or description,
 * positioned to the top relative to the target,
 * with customizable properties such as label, description, position, and maxWidth.
 *
 * The component utilizes the `useTooltip` hook, which simplifies the integration of tooltip behavior
 * by providing properties and methods required to manage the tooltip and its trigger seamlessly.
 *
 * The `useTooltip` hook provides two keys props:
 *
 * `getTriggerProps()`: Applied to the trigger element (e.g., a Button or an Icon, etc)
 * to bind the necessary event handlers for displaying the tooltip.
 *
 * `getTooltipProps()`: Applied to the Tooltip component to manage its visibility, positioning, and accessibility.
 */
export const BasicUsage = {
  args: {
    description: 'Tooltip text', // Default value for description
    label: 'Label', // Default value for label
  },
  // This is to style the story, as the story parent container has overflow hidden.
  decorators: [
    (Story) => (
      <div
        style={{
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          height: '20vh',
        }}
      >
        <Story />
      </div>
    ),
  ],
  render: (args: any) => {
    const tooltip = useTooltip()

    return (
      <>
        <Button {...tooltip.getTriggerProps()}>Hover me</Button>
        <Tooltip {...tooltip.getTooltipProps()} {...args} />
      </>
    )
  },
  parameters: {
    docs: {
      source: {
        code: `
render: () => {
 const tooltip = useTooltip()
 return (
    <Button {...tooltip.getTriggerProps()}>Hover me</Button>
    <Tooltip {...tooltip.getTooltipProps()} label="Label" description="This is a tooltip" />
  )
}
        `,
      },
    },
  },
}

// Mock createPortal for this story
const MockedTooltip = (props: TooltipProps) => {
  const content = (
    <ElTooltip role="tooltip" data-position={props.position} style={{ maxWidth: props.maxWidth }} aria-live="assertive">
      {props.label && <ElTooltipLabel>{props.label}: </ElTooltipLabel>}
      {props.description}
    </ElTooltip>
  )

  return content
}

/**
 * Below is the UI representation for the tooltip component without the trigger.
 */
export const DisplayTooltipWithoutTrigger = {
  args: {
    description: 'Tooltip text', // Default value for description
    label: 'Label', // Default value for label
    position: 'top',
    maxWidth: '400px',
    isVisible: true,
  },
  render: (args) => {
    console.log('args', args)
    // Pass args to MockedTooltip to make it reactive
    return <MockedTooltip {...args} />
  },
  parameters: {
    docs: {
      canvas: {
        sourceState: 'none',
      },
      story: {
        inline: false,
        iframeHeight: 60,
      },
    },
  },
}
