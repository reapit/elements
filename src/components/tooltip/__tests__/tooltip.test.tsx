import React from 'react'
import { vi } from 'vitest'
import { fireEvent, render } from '@testing-library/react'
import { useTooltip } from '../use-tooltip'
import { Button } from '../../button'
import { Tooltip } from '../tooltip'

// Mock createPortal for rendering Tooltip to the body
vi.mock('react-dom', async () => {
  const actual = await vi.importActual<typeof import('react-dom')>('react-dom')
  return {
    ...actual,
    createPortal: (node: React.ReactNode) => node, // Render the node directly
  }
})

describe('Tooltip with createPortal', () => {
  it('should render the tooltip when visible', () => {
    const TestComponent = () => {
      const tooltip = useTooltip()
      return (
        <>
          <Button {...tooltip.getTriggerProps()}>Hover me</Button>
          <Tooltip
            {...tooltip.getTooltipProps()}
            label="Label"
            description="This is a tooltip"
            isVisible={true}
            position="top"
            maxWidth="400px"
          />
        </>
      )
    }

    const { container, getByText } = render(<TestComponent />)
    // Trigger visibility
    const button = getByText('Hover me')
    fireEvent.mouseEnter(button)

    // Snapshot test with the tooltip visible
    expect(container).toMatchSnapshot()
  })

  it('should not render the tooltip when hidden', () => {
    const TestComponent = () => {
      const tooltip = useTooltip()
      return (
        <>
          <Button {...tooltip.getTriggerProps()}>Hover me</Button>
          <Tooltip
            {...tooltip.getTooltipProps()}
            label="Label"
            description="This is a tooltip"
            isVisible={false}
            position="top"
            maxWidth="400px"
          />
        </>
      )
    }

    const { container } = render(<TestComponent />)
    // Snapshot test with the tooltip hidden
    expect(container).toMatchSnapshot()
  })
})
