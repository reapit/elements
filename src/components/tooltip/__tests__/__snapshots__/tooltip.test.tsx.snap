// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip with createPortal > should not render the tooltip when hidden 1`] = `
<div>
  <button
    aria-describedby="tooltip-id-test-static-id"
    aria-disabled="false"
    class="el-button-size-medium el-button"
    data-visible-id="tooltip-id-test-static-id"
    role="button"
  >
    <div
      class="mocked-styled-0 el-button-spinner"
    />
    <span
      class="mocked-styled-1 el-button-label"
    >
      Hover me
    </span>
  </button>
</div>
`;

exports[`Tooltip with createPortal > should render the tooltip when visible 1`] = `
<div>
  <button
    aria-describedby="tooltip-id-test-static-id"
    aria-disabled="false"
    class="el-button-size-medium el-button"
    data-visible-id="tooltip-id-test-static-id"
    role="button"
  >
    <div
      class="mocked-styled-0 el-button-spinner"
    />
    <span
      class="mocked-styled-1 el-button-label"
    >
      Hover me
    </span>
  </button>
  <div
    aria-hidden="false"
    class="mocked-styled-7 el-tooltip"
    data-position="top"
    id="tooltip-id-test-static-id"
    role="tooltip"
    style="max-width: 400px; position: absolute; transform: translate(0px, -4px); margin: 0px;"
  >
    <span
      class="mocked-styled-8 el-tooltip-label"
    >
      Label
      : 
    </span>
    This is a tooltip
  </div>
</div>
`;
